contracts/src/Teleporter/upgrades/TeleporterRegistry.sol

```diff
@@ -37,6 +111,85 @@ contract TeleporterRegistry is WarpProtocolRegistry {
         view
         returns (ITeleporterMessenger)
     {
-        return ITeleporterMessenger(_getAddressFromVersion(_latestVersion));
+        return ITeleporterMessenger(getAddressFromVersion(latestVersion));
+    }
+
+    /**
+     * @dev Gets the {ITeleporterMessenger} contract of the given `version`.
+     * Requirements:
+     *
+     * - `version` must be a valid version, i.e. greater than 0 and not greater than the latest version.
+     */
+    function getTeleporterFromVersion(
+        uint256 version
+    ) external view returns (ITeleporterMessenger) {
+        return ITeleporterMessenger(getAddressFromVersion(version));
+    }
+
+    /**
+     * @dev Gets the address of a protocol version.
+     * Requirements:
+     *
+     * - `version` must be a valid version, i.e. greater than 0 and not greater than the latest version.
+     */
+    function getAddressFromVersion(
+        uint256 version
+    ) public view returns (address) {
+        require(version != 0, "TeleporterRegistry: zero version");
+        require(
```

Comment:
The current behavior here is to revert if `version > latestVersion`, but not revert otherwise if `version` isn't contained in `_versionToAddress`. I think we should have the same behavior in all cases the `version` isn't explicit set in `_versionToAddress`.

Reply:
good point, mentioned above but changed it to check for registered inputs, and to revert if not.